CANTroller2/src/devices.h

```diff
@@ -350,352 +351,37 @@ class Sensor : public Transducer<NATIVE_T, HUMAN_T> {
-                }
-                kv = _devices.find(option); // we need to set the input source of the newly-overridden component
-                if (kv != _devices.end()) {
-                    Device *d = std::get<1>(kv->second);
-                    if (d != nullptr) { // if  we're overloading a component with an associated devicie, we need to change the input source to the pot
-                        bool can_sim = std::get<0>(kv->second);
-                        if (can_sim) { // if we allow simualation for this componenent...
-                            d->set_source(ControllerMode::POT); // ...then set its input source to the pot
-                        }
-                    }
-                }
-                _pot_overload = option;
-            }
-        }
-
-        bool get_enabled() { return _enabled; }
-        SimOption get_pot_overload() { return _pot_overload; }
-};
-
 // class AnalogSensor are sensors where the value is based on an ADC reading (eg brake pressure, brake actuator position, pot)
 template<typename NATIVE_T, typename HUMAN_T>
 class AnalogSensor : public Sensor<NATIVE_T, HUMAN_T> {
   protected:
-    // NOTE: pot could be an actual Device as well...
-    std::shared_ptr<float> _pot_val; // to pull input from the pot if we're in simulation mode
-  public:
-    AnalogSensor(uint8_t arg_pin) : Sensor<NATIVE_T, HUMAN_T>(arg_pin) {}
-    void read() {
-        switch (this->_source) {
-            case ControllerMode::PIN: 
-                this->set_native(analogRead(this->_pin));
-                this->ema(this->human.get()); // filtered values are kept in human format
-                break;
-            case ControllerMode::TOUCH:
-                this->_val_filt.set(this->human.get());
-                break;
-            case ControllerMode::POT:
-                this->human.set(map(*this->_pot_val, Potentiometer::percent_min, Potentiometer::percent_max, this->human.get_min(), this->human.get_max()));
-                this->_val_filt.set(this->human.get());
-                break;
-        } // take no action in other modes
+    virtual void handle_pin_mode() {
+        this->set_native(static_cast<NATIVE_T>(analogRead(this->_pin)));
+        this->ema(this->human.get()); // filtered values are kept in human format
+    }
+    virtual void handle_touch_mode() {
+        this->_val_filt.set(this->human.get());
     }
-    void enable_pot_input(std::shared_ptr<float> pot_val_arg) {
-        this->set_can_source(ControllerMode::POT, true);
-        _pot_val = pot_val_arg;
+    virtual void handle_pot_mode() {
+        this->human.set(this->_pot->mapToRange(this->human.get_min(), this->human.get_max()));
+        this->_val_filt.set(this->human.get()); // don't filter the value we get from the pot, the pot output is already filtered
     }
+  public:
+    AnalogSensor(uint8_t arg_pin) : Sensor<NATIVE_T, HUMAN_T>(arg_pin) {}
 };
 
 // TODO: add description
```

Comment:
// PressureSensor represents a brake fluid pressure sensor.
// It extends AnalogSensor to handle reading an analog pin 
// and converting the ADC value to a pressure value in PSI.

Reply:
ty